### ISSUE TITLE
Layer2 problem with New-NsxFirewallRule ?

### ISSUE THREAD

**Issue by Brado231** [vmware/powernsx#244]
Hi,

Im running PowerCLI 6.5 Release 1 with the latest version of PowerNSX and can successfully set Layer 3 rules using New-NsxFirewallRule but cannot set Layer 2 rules. I'm running NSX 6.3.1.

The command I am trying to run is as follows:

Get-NsxFirewallSection "Default Section Layer2" | New-NsxFirewallRule -RuleType layer2sections -name BLOCK-INTRA-GOLD-TRAFFIC -Source (get-nsxsecuritygroup ($pod + "-GOLD")) -Destination (get-nsxsecuritygroup ($pod + "-GOLD")) -Action deny

I have tried creating a section under Layer 2 and adding to that but get the same failure result. I have also tried running a simple command like:

Get-NsxFirewallSection "Default Section Layer2" | New-NsxFirewallRule -RuleType layer2sections -name test -Action allow

...but this also fails.

At a PowerCLI prompt the commands seem to execute successfully with no errors, but nothing is returned and the rule is not visible in the NSX GUI.

Layer 3 rules work fine. It only seems like the creation of Layer 2 rules are affected.

Can someone please confirm if they also see the issue using NSX 6.3.1 or whether I am doing something wrong?

Thanks,
Brad

**Comment by nmbradford** [comment]
You are hitting some stupidness in the DFW API that I havent hidden sorry.  

Effectively, the L2, L3 and L3Redirect APIS are all separate, and when manipulating them, you have to specify as such.  Unfortunately, that means that when you call the get-nsxfirewallsection, you _must_ specify a section type of layer2sections.  

Without this param, it defaults to getting only L3 sections, of which none is named "Default Section Layer2", and the first part of your pipeline produces no objects and therefore, the second part is not executed at all (which is why you dont get an error - this is a pretty standard PowerShell gotcha.)

So - this should work for you : 
```
Get-NsxFirewallSection -name "Default Section Layer2" -sectiontype layer2sections | New-NsxFirewallRule -RuleType layer2sections -name test -Action allow
```

**Comment by nmbradford** [comment]
actually - ive just taken a closer gander at this and I think its worth a functionality enhancement request.  I think this can more intuitive without making it more expensive API query wise.

**Comment by nmbradford** [comment]
raised issue #249 to track some improvements to this pipeline.  Not sure your exact scenario is covered by it, but it bears some work without doubt.  thanks.  can I close the issue?

**Comment by Brado231** [comment]
Thanks Nick. That works perfectly now after including the -sectiontype switch.

Maybe we could have both layer 2 and layer 3 examples in "get-help get-nsxfirewallsection -examples" and "get-help new-nsxfirewallrule -examples"?

Happy to close this off. Thanks for your help!

**Comment by nmbradford** [comment]
fixed in #394
